vpr/src/route/clock_network_builders.cpp

```diff
@@ -343,7 +343,15 @@ int ClockRib::create_chanx_wire(int x_start,
     }
     node.set_cost_index(CHANX_COST_INDEX_START + seg_index); // Actual value set later
 
-    add_to_rr_node_indices(*rr_node_indices, *rr_nodes, node_index);
+    /* Add the node to spatial lookup */
+    auto& rr_graph = (*rr_nodes);
+    RRNodeId chanx_node = RRNodeId(node_index);
+    for (int ix = rr_graph.node_xlow(chanx_node); ix <= rr_graph.node_xhigh(chanx_node); ++ix) {
+        for (int iy = rr_graph.node_ylow(chanx_node); iy <= rr_graph.node_yhigh(chanx_node); ++iy) {
+            //TODO: CHANX uses odd swapped x/y indices here. Will rework once rr_node_indices is shadowed
+            rr_graph_builder.node_lookup().add_node(chanx_node, iy, ix, rr_graph.node_type(chanx_node), rr_graph.node_ptc_num(chanx_node), SIDES[0]);
+        }
+    }
 
     return node_index;
```

Comment:
Consider changing return type to RRNodeId, and directly making node_index an RRNodeId (and therefore wouldn't need chanx_node).

Reply:
Actually, I believe, we can use the API ``add_node_to_all_locs()`` as you suggested in another PR #1800. These for loops will be replaced by a simple line: 

```
rr_graph_builder.add_node_to_all_locs(chanx_node)
```

I leave a TODO comment here to remind me to finish the API replacement after PR #1800 is merged.
Let me know if this is fine for you.